lib/ansible/plugins/inventory/tower.py

```diff
@@ -0,0 +1,178 @@
+# Copyright (c) 2018 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: tower
+    plugin_type: inventory
+    authors:
+      - Matthew Jones (@matburt)
+      - Yunfan Zhang (@YunfanZhang42)
+    short_description: Ansible dynamic inventory plugin for Ansible Tower.
+    description:
+        - Reads inventories from Ansible Tower.
+        - Supports reading configuration from both YAML config file and environment variables.
+        - If reading from the YAML file, the file name must end with tower_inventory.(yml|yaml),
+          the path in the command would be /path/to/tower_inventory.(yml|yaml). If some arguments in the config file
+          are missing, this plugin will try to fill in missing arguments by reading from environment variables.
+        - If reading configurations from environment variables, the path in the command must be @tower_inventory.
+    options:
+        tower_host:
+            description: The network address of your Ansible Tower host.
+            type: string
+            env:
+                - name: TOWER_HOST
+            required: True
```

Comment:
How is this required if it can also be specified by the environment variable? Compare to openshift plugin via `ansible-doc -t inventory tower`

```
api_key:
  description:
  - Token used to authenticate with the API. Can also be specified via K8S_AUTH_API_KEY
    environment variable.
```

Reply:
My understanding is this: openshift uses its own method in their own pip packages to read from environment variables and check if the option is either supplied with env vars or yml. 

Our plugin uses Ansible's parser to read from env vars and/or yml, so we have to set `required`. I tested with not setting `required` at all and Ansible's parser will not enforce the `required` behavior.

Reply:
required doesn't mean 'requried in config', it  means a value MUST be provided, any way you allow to set it is fine